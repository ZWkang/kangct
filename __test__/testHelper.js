// import React from 'react';
import { mount, shallow, render } from 'enzyme';

function setUp(type, ...arg) {
  const [Component, props = {}, children = ''] = arg;
  if (~['mount', 'shallow', 'render'].indexOf(type)) {
    return global[type](<Component {...props}>{children}</Component>);
  }
}

export function AutoMath(Component, props = {}, children = '') {
  const wrapper = setUp('render', Component, props, children);
  expect(wrapper).toMatchSnapshot();
}

export function ComponentRenderWithoutError(Component, props = {}, children = '') {
  const wrapper = setUp('mount', Component, props, children);
  expect(() => {
    wrapper.setProps({});
    wrapper.unmount();
  }).not.toThrow();
}

export function childrenShouldBeProps(Component, children) {
  const wrapper = setUp('mount', Component, {}, children);

  expect(wrapper.get(0).props.children).toEqual(children);
}

export function setup(Component, props) {
  const wrapper = setUp(Component, props, null);
  return {
    wrapper,
    props
  };
}
