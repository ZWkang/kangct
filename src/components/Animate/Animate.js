import styled, { keyframes } from 'styled-components';
import PropTypes from 'prop-types';
import animated from './help.js';
import React, { Component } from 'react';
// import omit from 'lodash.omit';
import { Object_omit as omit } from '../util/index.js';

const generatorComponent = ({ name, duration, keyframesFuntion, delay }) => {
  const animateKeyframs = keyframes`${keyframesFuntion || animated[name]}`;
  const Wrapper = styled.div`
    will-change: contents;
    animation: ${animateKeyframs};
    animation-duration: ${duration || 0}ms;
    animation-delay: ${delay || 0}ms;
  `;
  return Wrapper;
};

class Animate extends Component {
  constructor(props) {
    super(props);
    this.RenderComponent = generatorComponent({
      keyframesFuntion: props.keyframesFuntion,
      name: props.funcname,
      duration: props.duration,
      delay: props.delay
    });
  }
  render() {
    const childProps = Object.assign({}, this.props);
    const omitProps = omit(childProps, [
      'funcname',
      'children',
      'name',
      'duration',
      'keyframesFuntion',
      'delay'
    ]);
    const { RenderComponent: Wrapper } = this;
    const Children = React.cloneElement(this.props.children, {
      ...omitProps
    });

    return <Wrapper>{Children}</Wrapper>;
  }
}

Animate.propTypes = {
  keyframesFuntion: PropTypes.func,
  name: PropTypes.string,
  duration: PropTypes.number
};

Animate.defaultProps = {
  keyframesFuntion: void 666,
  name: '',
  duration: 0
};
/**
 * @components
 */

export default Animate;
