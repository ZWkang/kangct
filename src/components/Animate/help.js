import {
  bouceOut,
  bounce,
  bounceIn,
  bounceInDown,
  bounceInLeft,
  bounceInRight,
  bounceInUp,
  bounceOutDown,
  bounceOutLeft,
  bounceOutRight,
  bounceOutUp,
  fadeIn,
  fadeInDown,
  fadeInDownBig,
  fadeInLeft,
  fadeInLeftBig,
  fadeInRight,
  fadeInRightBig,
  fadeInUp,
  fadeInUpBig,
  fadeOut,
  fadeOutDown,
  fadeOutDownBig,
  fadeOutLeft,
  fadeOutLeftBig,
  fadeOutRight,
  fadeOutRightBig,
  fadeOutUp,
  fadeOutUpBig,
  flash,
  flip,
  flipInX,
  flipInY,
  flipOutX,
  flipOutY,
  headShake,
  hinge,
  jello,
  lightSpeedIn,
  lightSpeedOut,
  pulse,
  rollIn,
  rollOut,
  rotateIn,
  rotateInDownLeft,
  rotateInDownRight,
  rotateInUpLeft,
  rotateInUpRight,
  rotateOut,
  rotateOutDownLeft,
  rotateOutDownRight,
  rotateOutUpLeft,
  rotateOutUpRight,
  rubberBand,
  shake,
  slideInDown,
  slideInLeft,
  slideInRight,
  slideInUp,
  slideOutDown,
  slideOutLeft,
  slideOutRight,
  slideOutUp,
  swing,
  tada,
  wobble,
  zoomIn,
  zoomInDown,
  zoomInLeft,
  zoomInRight,
  zoomInUp,
  zoomOut,
  zoomOutDown,
  zoomOutLeft,
  zoomOutRight,
  zoomOutUp,
  merge
} from 'react-animations';

export {
  bouceOut,
  bounce,
  bounceIn,
  bounceInDown,
  bounceInLeft,
  bounceInRight,
  bounceInUp,
  bounceOutDown,
  bounceOutLeft,
  bounceOutRight,
  bounceOutUp,
  fadeIn,
  fadeInDown,
  fadeInDownBig,
  fadeInLeft,
  fadeInLeftBig,
  fadeInRight,
  fadeInRightBig,
  fadeInUp,
  fadeInUpBig,
  fadeOut,
  fadeOutDown,
  fadeOutDownBig,
  fadeOutLeft,
  fadeOutLeftBig,
  fadeOutRight,
  fadeOutRightBig,
  fadeOutUp,
  fadeOutUpBig,
  flash,
  flip,
  flipInX,
  flipInY,
  flipOutX,
  flipOutY,
  headShake,
  hinge,
  jello,
  lightSpeedIn,
  lightSpeedOut,
  pulse,
  rollIn,
  rollOut,
  rotateIn,
  rotateInDownLeft,
  rotateInDownRight,
  rotateInUpLeft,
  rotateInUpRight,
  rotateOut,
  rotateOutDownLeft,
  rotateOutDownRight,
  rotateOutUpLeft,
  rotateOutUpRight,
  rubberBand,
  shake,
  slideInDown,
  slideInLeft,
  slideInRight,
  slideInUp,
  slideOutDown,
  slideOutLeft,
  slideOutRight,
  slideOutUp,
  swing,
  tada,
  wobble,
  zoomIn,
  zoomInDown,
  zoomInLeft,
  zoomInRight,
  zoomInUp,
  zoomOut,
  zoomOutDown,
  zoomOutLeft,
  zoomOutRight,
  zoomOutUp,
  merge
};
export default {
  bouceOut,
  bounce,
  bounceIn,
  bounceInDown,
  bounceInLeft,
  bounceInRight,
  bounceInUp,
  bounceOutDown,
  bounceOutLeft,
  bounceOutRight,
  bounceOutUp,
  fadeIn,
  fadeInDown,
  fadeInDownBig,
  fadeInLeft,
  fadeInLeftBig,
  fadeInRight,
  fadeInRightBig,
  fadeInUp,
  fadeInUpBig,
  fadeOut,
  fadeOutDown,
  fadeOutDownBig,
  fadeOutLeft,
  fadeOutLeftBig,
  fadeOutRight,
  fadeOutRightBig,
  fadeOutUp,
  fadeOutUpBig,
  flash,
  flip,
  flipInX,
  flipInY,
  flipOutX,
  flipOutY,
  headShake,
  hinge,
  jello,
  lightSpeedIn,
  lightSpeedOut,
  pulse,
  rollIn,
  rollOut,
  rotateIn,
  rotateInDownLeft,
  rotateInDownRight,
  rotateInUpLeft,
  rotateInUpRight,
  rotateOut,
  rotateOutDownLeft,
  rotateOutDownRight,
  rotateOutUpLeft,
  rotateOutUpRight,
  rubberBand,
  shake,
  slideInDown,
  slideInLeft,
  slideInRight,
  slideInUp,
  slideOutDown,
  slideOutLeft,
  slideOutRight,
  slideOutUp,
  swing,
  tada,
  wobble,
  zoomIn,
  zoomInDown,
  zoomInLeft,
  zoomInRight,
  zoomInUp,
  zoomOut,
  zoomOutDown,
  zoomOutLeft,
  zoomOutRight,
  zoomOutUp,
  merge
};
