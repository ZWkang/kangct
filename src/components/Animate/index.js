import styled, { keyframes } from 'styled-components';
import { bounce } from 'react-animations';

const bounceAnimation = keyframes`${bounce}`;

const BouncyDiv = styled.div`
  animation: 1s ${bounceAnimation};
`;


export default BouncyDiv