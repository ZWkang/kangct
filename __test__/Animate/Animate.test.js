import React from 'react';
import { shallow, mount } from 'enzyme';
import Animate from '@app/Animate/index.js';
import { AutoMath, ComponentRenderWithoutError, childrenShouldBeProps } from '../testHelper';
import 'jest-styled-components';

function noop() {}
function createAnimateProps({
  funcName = 'bounce',
  duration = 2000,
  delay = 0,
  onAnimationEnd = noop,
  keyframesFunction = noop
}) {
  return {
    funcName,
    duration,
    delay,
    onAnimationEnd,
    keyframesFunction
  };
}
// 分成对应的describe
// 1. 测试渲染与快照
// 2. 测试UI控制，props传递，以及实际交互
// 3. 生命周期调用是否生效。

// 你希望在重构时候检查什么，就在这里写什么测试
describe('Animate Component', () => {
  it('render without error', () => {
    ComponentRenderWithoutError(Animate, {}, <p>asda</p>);
  });
  it('match snapshot', () => {
    AutoMath(Animate, { funcName: 'bounce', duration: 3000 }, <h1>here is a simple test</h1>);
  });
  it('test animation work?', () => {
    jest.useFakeTimers();

    const animationend = jest.fn();

    const wrapperProps = createAnimateProps({ onAnimationEnd: animationend });
    const wrapper = mount(
      <Animate {...wrapperProps}>
        <p>test</p>
      </Animate>
    );
    wrapper.simulate('animationend');
    expect(animationend).toBeCalled();
  });
  it('test more function name', () => {
    const bounceAnimationProps = createAnimateProps({ funcName: 'bounce' });

    const bounceAnimation = mount(<Animate {...bounceAnimationProps}>test</Animate>);
    expect(bounceAnimation.text()).toEqual('test');

    // console.log(bounceAnimation.debug());
    // console.log(bounceAnimation.get(0));
    expect(bounceAnimation.get(0).props.funcName).toEqual('bounce');
    const bounceInAnimationProps = createAnimateProps({ funcName: 'bounceIn' });

    const bounceInAnimation = mount(<Animate {...bounceInAnimationProps}>test</Animate>);
    expect(bounceInAnimation.text()).toEqual('test');

    expect(bounceInAnimation.get(0).props.funcName).toEqual('bounceIn');
  });

  it('children should be props', () => {
    childrenShouldBeProps(Animate, <p>test</p>);
  });
});
