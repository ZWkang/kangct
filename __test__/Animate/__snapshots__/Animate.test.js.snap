// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animate Component match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bwzfXH irmJKN",
    },
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "here is a simple test",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h1",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
